src/artwork.c

```diff
@@ -218,6 +226,57 @@ static struct artwork_source artwork_item_source[] =
 
 /* -------------------------------- HELPERS -------------------------------- */
 
+/* Reads an artwork file from the given url straight into an evbuf
+ *
+ * @out evbuf     Image data
+ * @in  url       URL for the image
+ * @return        0 on success, -1 on error
+ */
+static int
+artwork_url_read(struct evbuffer *evbuf, const char *url)
+{
+  struct http_client_ctx client;
+  struct keyval *kv;
+  const char *content_type;
+  int len;
+  int ret;
+
+  DPRINTF(E_SPAM, L_ART, "Trying internet artwork in %s\n", url);
+
+  ret = ART_E_NONE;
+
+  len = strlen(url);
+  if ((len < 14) || (len > PATH_MAX)) // Can't be shorter than http://a/1.jpg
+    goto out_url;
+
+  kv = keyval_alloc();
+  if (!kv)
+    goto out_url;
+
+  memset(&client, 0, sizeof(struct http_client_ctx));
+  client.url = url;
+  client.input_headers = kv;
+  client.input_body = evbuf;
+
+  if (http_client_request(&client) < 0)
+    goto out_kv;
+
+  content_type = keyval_get(kv, "Content-Type");
+  if (content_type && (strcmp(content_type, "image/jpeg") == 0))
+    ret = ART_FMT_JPEG;
+  else if (content_type && (strcmp(content_type, "image/png") == 0))
+    ret = ART_FMT_PNG;
+  else
+    ret = ART_FMT_JPEG;
```

Comment:
Does Spotify not set the content type? If it does I would remove this... otherwise we might end up serving stuff like html 404's as jpg.

Reply:
Spotify does send the content type header, but our `http_client_request` function did not store them in the key-value store (that was only done for http requests and not for https requests). 

I modified `http_client_request` to

- store the response code in the ctx struct
- add the content type header to the key-value store for requests made with libcurl

`artwork_url_read` now checks the response code and the content type.